### PR TITLE
Remove bs4

### PR DESCRIPTION
**Pull request by zhou-pj**
I changed the text renderer using the parser within the mistune to remove the bs4, and add tabulate.py in the plugins to address mistune owner's comments.

### CODE REVIEW

File: mistune/plugins/table.py [hunk 1]
```diff
@@ -27,6 +27,17 @@ def parse_table(self, m, state):
     children = [thead, {'type': 'table_body', 'children': rows}]
     return {'type': 'table', 'children': children}
 
+def text_parse_and_render_table(self, m, state):
+    table = parse_table(self, m, state)
+
+    from .tabulate import tabulate
```
Comment (csadorf) [inline mistune/plugins/table.py]: Should we make that a module-level import?

File: mistune/plugins/tabulate.py [hunk 2]
```diff
@@ -0,0 +1,1772 @@
+# -*- coding: utf-8 -*-
```
Comment (csadorf) [inline mistune/plugins/tabulate.py:1]: I assume we need to include the *tabulate* license if we bundle this module?

File: mistune/renderers.py [hunk 3]
```diff
@@ -245,19 +245,10 @@ def codespan(self, text):
...
-        from tabulate import tabulate
-        soup = BeautifulSoup('<table>' + table + '</table>', 'html.parser')
-        table = soup.find('table')
-        thead = table.find('thead')
-        headers = [cell.text for cell in thead.find_all('th')]
-        tbody = table.find('tbody')
-        rows = [[cell.text for cell in row.find_all('td')]
-                 for row in tbody.find_all('tr')]
-        return tabulate(rows, headers=headers) + self.linebreak() + self.linebreak()
+        return '\033[1m' + text + '\033[0m'
```
Comment (csadorf) [inline mistune/renderers.py]: That change seems unrelated?

**Review by csadorf — COMMENTED** [review]
Thank you very much for taking this on!

File: mistune/plugins/tabulate.py (hunk 2 above)
Comment (zhou-pj) [reply mistune/plugins/tabulate.py:1]: I added the license doc as docstring in tabulate.py, is this the proper way to include it?

File: mistune/renderers.py (hunk 3 above)
Comment (zhou-pj) [reply mistune/renderers.py]: This change is necessary to yield proper bold setting for terminal output. Would you prefer a separate pull request for this change?

File: mistune/plugins/tabulate.py (hunk 2 above)
Comment (csadorf) [reply mistune/plugins/tabulate.py:1]: That's probably sufficient.

**Comment by zhou-pj** [comment]
@csadorf Do you want to have another look at this PR?

**Review by csadorf — APPROVED** [review]
Looks good!
